client/src/components/Map.tsx

```diff
@@ -120,18 +111,59 @@ const MapComponent: FC<MapComponentProps> = (props) => {
       clearMarkers();
       return;
     }
+
     const newMarkers: google.maps.marker.AdvancedMarkerElement[] = documents
       .filter((doc) => doc.coordinates)
       .filter((doc) => (visualLinks ? isSelectedOrLinked(doc) : true))
       .map((doc) =>
         createMarker(doc, visualLinks && doc.id !== documentSelected?.id)
       );
 
-    setMarkers((_) => {
-      clearMarkers();
-      return newMarkers;
+    clearMarkers();
+
+    const markerCluster = new MarkerClusterer({
+      markers: newMarkers,
+      map,
+      // Usa il `renderer` per personalizzare l'icona dei cluster
+      renderer: {
+        render: ({ count, position }) => {
+          return new google.maps.marker.AdvancedMarkerElement({
+            position,
+            title: count.toString(),
+            content: (() => {
+              const div = document.createElement("div");
+              div.className = "cluster-icon";
+              div.textContent = count.toString();
+              div.style.textAlign = "center";
```

Comment:
Put the styles in the scss not in the components

Reply:
Done bro